tools/ci/choose_workflows.py

```diff
@@ -0,0 +1,166 @@
+) -> bool:
+    """
+    Resolves CI do determine if a phase should be ran as part of the workflow.
+
+    :param ctx: CI context (ENVs and optional Pull Request information)
+    :param trigger_env: the name of a trigger-level ENV which decides on running this phase
+    :param build_env: the name of a build-level ENV which decides on running this phase
+    :param pr_keyword: the magic word to lookup in PR's description (or `None`) - if found, it will trigger this phase
+    :param pr_path_prefixes: the list of prefixes to match against PR's modified files (any match triggers this phase)
+    :param pr_file_extensions: the list of suffixes to match against PR's modified files (any match triggers this phase)
+    :return: True if a phase should be ran for given CI context
+    """
+    # First, respect trigger ENV:
+    if trigger_env == '1':
+        print('→ opted-in by trigger ENV')
+        return True
+
+    # Second, check build ENV:
+    if build_env == '1':
+        print('→ opted-in by build ENV')
+        return True
+
+    # Last, infer from Pull Request (if running for PR):
+    if ctx.pull_request:
+        if pr_keyword and ctx.pull_request.description.contains(pr_keyword):
+            print(f'→ opted-in by matching keyword ("{pr_keyword}") in PR description ▶️')
+            return True
+
+        if ctx.pull_request.modified_files.contains_paths(path_prefixes=pr_path_prefixes):
+            print(f'→ opted-in by matching one or more path prefixes ({pr_path_prefixes}) in PR files ▶️')
+            return True
+
+        if ctx.pull_request.modified_files.contains_extensions(file_extensions=pr_file_extensions):
+            print(f'→ opted-in by matching one or more file extensions ({pr_file_extensions}) in PR files ▶️')
+            return True
+
+    print(f'→ will be skipped ⏩')
+    return False
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dry-run",
+        action='store_true',
+        help="Run without exporting ENVs to `envman`."
+    )
+    args = parser.parse_args()
+
+    try:
+        dry_run = True if args.dry_run else False
+
+        context = get_ci_context()
+
+        print(f'🛠️️ CI Context:\n'
+              f'- trigger ENVs = {context.trigger_env}\n'
+              f'- build ENVs   = {context.build_env}\n'
+              f'- PR files     = {context.pull_request.modified_files.paths if context.pull_request else "(no PR)"}')
+
+        if should_run_unit_tests(context):
+            export_env('DD_RUN_UNIT_TESTS', '1', dry_run=dry_run)
+
+        if should_run_integration_tests(context):
+            export_env('DD_RUN_INTEGRATION_TESTS', '1', dry_run=dry_run)
+
+        if should_run_smoke_tests(context):
+            export_env('DD_RUN_SMOKE_TESTS', '1', dry_run=dry_run)
+
+        if should_run_tools_tests(context):
+            export_env('DD_RUN_TOOLS_TESTS', '1', dry_run=dry_run)
+
```

Comment:
I also wonder on reporting enabled phases back to PR (if running for pull request). It can be done easily with GH CLI and sending a PR comment, e.g.:

<img width="907" alt="Screenshot 2022-04-21 at 08 40 27" src="https://user-images.githubusercontent.com/2358722/164390104-ad0b4e52-3d1a-4b79-b6d4-1c7ce6ae12c3.png">

This would be only a sanity measure - reporting which phases are going to be ran, without reporting their status (this is done by Bitrise syncing back to GH Checks). Not sure if this would be helpful / annoying - WDYT @maxep?

Reply:
I don't think that's necessary, we have GH checks that already gives us that info 👍

Reply:
Makes sense 👍.